homeassistant/components/linknlink/config_flow.py

```diff
@@ -0,0 +1,271 @@
+            return self.async_show_form(
+                step_id="reset",
+                errors=errors,
+                description_placeholders={
+                    "name": device.name,
+                    "model": device.model,
+                    "host": device.host[0],
+                },
+            )
+
+        return await self.async_step_user(
+            {CONF_HOST: device.host[0], CONF_TIMEOUT: device.timeout}
+        )
+
+    async def async_step_unlock(self, user_input=None) -> FlowResult:
+        """Unlock the device.
+
+        The authentication succeeded, but the device is locked.
+        We can offer an unlock to prevent authorization errors.
+        """
+        device = self.device
+        errors = {}
+
+        if user_input is None:
+            pass
+
+        elif user_input["unlock"]:
+            try:
+                await self.hass.async_add_executor_job(device.set_lock, False)
+
+            except NetworkTimeoutError as err:
+                errors["base"] = "cannot_connect"
+                err_msg = str(err)
+
+            except LinknLinkException as err:
+                errors["base"] = "unknown"
+                err_msg = str(err)
+
+            except OSError as err:
+                if err.errno == errno.ENETUNREACH:
+                    errors["base"] = "cannot_connect"
+                    err_msg = str(err)
+                else:
+                    errors["base"] = "unknown"
+                    err_msg = str(err)
+
+            else:
+                return await self.async_step_finish()
+
+            _LOGGER.error(
+                "Failed to unlock the device at %s: %s", device.host[0], err_msg
+            )
+
+        else:
+            return await self.async_step_finish()
+
+        data_schema = {vol.Required("unlock", default=False): bool}
+        return self.async_show_form(
+            step_id="unlock",
+            errors=errors,
+            data_schema=vol.Schema(data_schema),
+            description_placeholders={
+                "name": device.name,
+                "model": device.model,
+                "host": device.host[0],
+            },
+        )
+
+    async def async_step_finish(self, user_input=None) -> FlowResult:
+        """Choose a name for the device and create config entry."""
+        device = self.device
+
+        # Abort reauthentication flow.
+        self._abort_if_unique_id_configured(
+            updates={CONF_HOST: device.host[0], CONF_TIMEOUT: device.timeout}
+        )
+
+        if user_input is not None:
+            return self.async_create_entry(
+                title=user_input[CONF_NAME],
+                data={
+                    CONF_HOST: device.host[0],
+                    CONF_MAC: device.mac.hex(),
+                    CONF_TYPE: device.devtype,
+                    CONF_TIMEOUT: device.timeout,
+                },
+            )
+
+        data_schema = {vol.Required(CONF_NAME, default=device.name): str}
```

Comment:
Only helpers are allowed to set a custom name. We should set the name to something the user can recognize